ouroboros-consensus/test/storage-test/Test/Ouroboros/Storage/LedgerDB/StateMachine.hs

```diff
@@ -0,0 +1,529 @@
+{-# LANGUAGE CPP                        #-}
+{-# LANGUAGE ConstraintKinds            #-}
+{-# LANGUAGE DataKinds                  #-}
+{-# LANGUAGE DeriveAnyClass             #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE DeriveTraversable          #-}
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE ExistentialQuantification  #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE GADTs                      #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE InstanceSigs               #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE NamedFieldPuns             #-}
+{-# LANGUAGE PackageImports             #-}
+{-# LANGUAGE PolyKinds                  #-}
+{-# LANGUAGE RankNTypes                 #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE StandaloneDeriving         #-}
+{-# LANGUAGE StandaloneKindSignatures   #-}
+{-# LANGUAGE TupleSections              #-}
+{-# LANGUAGE TypeApplications           #-}
+{-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE TypeOperators              #-}
+{-# LANGUAGE UndecidableInstances       #-}
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+#if __GLASGOW_HASKELL__ >= 908
+{-# OPTIONS_GHC -Wno-x-partial #-}
+#endif
+
+-- | On-disk ledger DB tests.
+--
+-- This is a model based test. The commands here are
+--
+-- * Get the current ledger state
+-- * Push a block, or switch to a fork
+-- * Write a snapshot to disk
+-- * Restore the ledger DB from the snapshots on disk
+-- * Model disk corruption
+--
+-- The model here is satisfyingly simple: just a map from blocks to their
+-- corresponding ledger state.
+--
+module Test.Ouroboros.Storage.LedgerDB.StateMachine (tests) where
```

Comment:
It would be useful to port the documentation we previously had on the `OnDisk` module to this module. I find it hard to judge whether the tests are exhaustive enough without

Reply:
Most of them do not apply anymore. They were subtleties that we had to introduce to make the model follow the implementation which (imho) is not what we should test here.

Reply:
Alright, but at least we would have to write some new documentation